src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -618,6 +618,9 @@ public async Task Initialize(HostWaitLoop waitLoop, CancellationToken token)
                         }
                     }
                 }
+                // now the exe is loaded and we can check target features
+                TargetFeatures = await MICommandFactory.GetTargetFeatures();
+                // TODO: only now we know if (TargetFeatures.Contains("async")), see above
```

Comment:
What to do about this `async` feature?

Reply:
We had an implementation in https://github.com/microsoft/MIEngine/pull/244 but had to revert with https://github.com/microsoft/MIEngine/pull/794 because of:

> This is being reverted as it hangs in the VS & Android case. gdb for Android supports async but doesnt' list it as capabilities. It is noticed when trying to stop debugging while running.

Reply:
See my comment there. It was simply done too early. I ran into the same problem.
So I guess `EnableTargetAsyncOption` could be moved to this line.